submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,211 @@
+        stack_size = items.raw[item]["stack_size"]
+        combinator.set_signal(signal_index, item, stack_size)
+        signal_index += 1
+        # Once we exceed the number of signals a combinator can hold, place it and reset
+        if signal_index == combinator.item_slot_count:
+            # Add the combinator to the blueprint
+            combinator.id = "{}_{}".format(x, y)
+            blueprint.entities.append(combinator)
+            # Reset the combinator
+            combinators_added += 1
+            y = combinators_added % COMBINATOR_HEIGHT
+            x = int(combinators_added / COMBINATOR_HEIGHT)
+            combinator.set_signals(None)  # Clear signals
+            combinator.tile_position = (x, y)
+            signal_index = 0
+
+    # Add the last combinator if partially full
+    combinator.id = "{}_{}".format(x, y)
+    blueprint.entities.append(combinator)
+
+    # Add connections to each neighbour
+    for cx in range(x):
+        for cy in range(COMBINATOR_HEIGHT):
+            here = "{}_{}".format(cx, cy)
+            right = "{}_{}".format(cx + 1, cy)
+            below = "{}_{}".format(cx, cy + 1)
+            try:
+                blueprint.add_circuit_connection("red", here, right)
+            except KeyError:
+                pass
+            try:
+                blueprint.add_circuit_connection("red", here, below)
+            except KeyError:
+                pass
+
+    print("Number of items added:", signal_count)
+    print(blueprint.to_string())
+
+
+if __name__ == "__main__":
+    main()
+```
+
+This script is concise and easy to follow, but the really amazing thing about Draftsman is that this script is completely dynamic and can adjust to whatever version of Factorio that you're playing. New items, removed items, or changed stack sizes, regardless of the author of the change, are resolved automatically with the exact same code. For illustration, pictured top-left is what's generated by running the script for a vanilla game, top-right a game with the medium-sized overhaul modpack [Space Exploration](https://mods.factorio.com/mod/space-exploration), and at the bottom a [Bobs](https://mods.factorio.com/user/Bobingabout) + [Pyanodon](https://mods.factorio.com/user/pyanodon) mega-modpack:
+
+![Results](./img/item_stack_size_output.png)
+
+This doesn't only applies to items either. All entities, instruments, signals, recipes, modules, and tiles are pulled from the emulated load process and then stored as *pickle* files in Draftsman for later use. Any script can be designed to be completely flexible across these categories; extra instruments in a new programmmable speaker, new module types only in certain machines, complete virtual signal lists for signal mappings, etc., all are interpreted correctly by Draftsman. By saving the data of the current mod configuration internally for later, this also means that you only have to update the data once, each time you change what mods you're using.
```

Comment:
I'd either clarify on what a pickle file is, or omit the mention of it entirely, since most readers won't know what it is.

Reply:
Feel free to omit it; can be rewritten as `and then stored in Draftsman for later use.`

Reply:
Unresolving until the changes are made in the article.